web-client/integration-tests/adcClerkViewsMessagesJourney.test.js

```diff
@@ -0,0 +1,296 @@
+    createNewMessageOnCase(cerebralTest, {
+      subject: message6Subject,
+      toSection: 'adc',
+      toUserId: testAdcId,
+    });
+    it('get message6SubjectFromState', () => {
+      message6SubjectFromState = cerebralTest.testMessageSubject;
+    });
+
+    createNewMessageOnCase(cerebralTest, {
+      subject: message7Subject,
+      toSection: 'adc',
+      toUserId: testAdcId,
+    });
+    it('get message7SubjectFromState', () => {
+      message7SubjectFromState = cerebralTest.testMessageSubject;
+    });
+
+    loginAs(cerebralTest, 'adc@example.com');
+    it('adc clerk completes message threads', async () => {
+      await cerebralTest.runSequence('gotoMessagesSequence', {
+        box: 'inbox',
+        queue: messageQueue,
+      });
+
+      const messages = cerebralTest.getState('messages');
+
+      await markMessageAsComplete(
+        cerebralTest,
+        messages,
+        message6Subject,
+        message6SubjectFromState,
+      );
+
+      await markMessageAsComplete(
+        cerebralTest,
+        messages,
+        message7Subject,
+        message7SubjectFromState,
+      );
+
+      await refreshElasticsearchIndex();
+    });
+
+    it(`verify default sorting of ${messageQueue} completed box completedAt sort field, descending`, async () => {
+      const { completedMessages } = await getUserMessageCount(
+        cerebralTest,
+        'completed',
+        messageQueue,
+      );
+
+      let afterCompletedMessageCount = cerebralTest.getState('messages').length;
+
+      const expected = [message7SubjectFromState, message6SubjectFromState];
+
+      expect(afterCompletedMessageCount).toEqual(
+        expected.length + beforeCompletedMessageCount,
+      );
+
+      validateMessageOrdering(completedMessages, expected);
+    });
+
+    loginAs(cerebralTest, 'docketclerk@example.com');
+    it('verify the table headers are not clickable for non-ADC users', async () => {
+      const showSortableHeaders = withAppContextDecorator(
+        showSortableHeadersComputed,
+      );
+
+      await cerebralTest.runSequence('gotoMessagesSequence', {
+        box: 'inbox',
+        queue: messageQueue,
+      });
+
+      const isTableHeaderClickable = runCompute(showSortableHeaders, {
+        state: cerebralTest.getState(),
+      });
+
+      expect(isTableHeaderClickable).toBe(false);
+    });
+    it('verify the messages in sent box are sorted from newest to oldest non-ADC users', async () => {
+      await cerebralTest.runSequence('gotoMessagesSequence', {
+        box: 'outbox',
+        queue: messageQueue,
+      });
+
+      const { messages } = runCompute(formattedMessagesComputed, {
+        state: cerebralTest.getState(),
+      });
+
+      const newSortedMessages = [...messages].sort((a, b) =>
+        b.createdAt.localeCompare(a.createdAt),
+      );
+
+      expect(messages).toEqual(newSortedMessages);
+    });
+  });
+});
+
+const validateMessageOrdering = (actualMessages, expectedMessageSubjects) => {
```

Comment:
💭 Could we make use of the `expect.arrayContaining` method in jest to enumerate what we expect the message ordering to be? That way if the test fails because the messages are not ordered as we expect, it may be more obvious why the sorting is not as we expected.

Reply:
After some research, we think expect.arrayContaining guarantees order. Part of the tests we performed was altering the ideal order and the tests returned a false positive.